src/platform/intel/cavs/include/cavs/lib/pm_runtime.h

```diff
@@ -16,12 +16,17 @@
 #ifndef __CAVS_LIB_PM_RUNTIME_H__
 #define __CAVS_LIB_PM_RUNTIME_H__
 
+#define PWRD_MASK	MASK(31, 30)
+#define PWRD_BY_HPRO	BIT(31)
+#define PWRD_BY_TPLG	BIT(30)
+
 struct pm_runtime_data;
 
 /** \brief cAVS specific runtime power management data. */
 struct cavs_pm_runtime_data {
 	int dsp_d0_sref; /**< simple ref counter, accessed by core 0 only */
 	int host_dma_l1_sref; /**< ref counter for Host DMA accesses */
+	int dsp_client_bitmap[PLATFORM_CORE_COUNT];	/**< simple pwr lock */
```

Comment:
why not use a proper lock ?

Reply:
@lgirdwood "lock" in the comment might not be the best name. I understand this rather as an "override" (by multiple clients).

Reply:
I'm not the best at inventing names and comments but this mechanism is something like "power lock", we can power up a core and lock it for example with HPRO flag so it can be powered down (unlocked) only with the same flag

Reply:
updated with override